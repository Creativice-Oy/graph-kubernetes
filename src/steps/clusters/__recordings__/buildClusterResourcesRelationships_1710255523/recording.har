{
  "log": {
    "_recordingName": "buildClusterResourcesRelationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "9d73307742273594a28ec2e4efce503e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 148,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "50"
            }
          ],
          "url": "https://192.168.49.2:8443/api/v1/namespaces?limit=50"
        },
        "response": {
          "bodySize": 1867,
          "content": {
            "mimeType": "application/json",
            "size": 1867,
            "text": "{\"kind\":\"NamespaceList\",\"apiVersion\":\"v1\",\"metadata\":{\"selfLink\":\"/api/v1/namespaces\",\"resourceVersion\":\"9161772\"},\"items\":[{\"metadata\":{\"name\":\"default\",\"selfLink\":\"/api/v1/namespaces/default\",\"uid\":\"9c533ce3-b6bd-43e6-947a-f9f73dc297b5\",\"resourceVersion\":\"155\",\"creationTimestamp\":\"2021-05-13T19:34:38Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:38Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-node-lease\",\"selfLink\":\"/api/v1/namespaces/kube-node-lease\",\"uid\":\"11a8778d-4ef0-437b-bb1d-db3cf0a26e82\",\"resourceVersion\":\"33\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-public\",\"selfLink\":\"/api/v1/namespaces/kube-public\",\"uid\":\"ba4e6047-864a-46f9-9f29-3ffe004d7f13\",\"resourceVersion\":\"25\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-system\",\"selfLink\":\"/api/v1/namespaces/kube-system\",\"uid\":\"de2a4eee-505b-49d7-bc08-09026c88dae5\",\"resourceVersion\":\"16\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 15:16:26 GMT"
            },
            {
              "name": "content-length",
              "value": "1867"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 146,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T15:16:26.942Z",
        "time": 25,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 25
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
