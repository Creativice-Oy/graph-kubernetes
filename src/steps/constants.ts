import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export const CLUSTER_ENTITY_DATA_KEY = 'entity:cluster';

export enum IntegrationSteps {
  NETWORK_POLICIES = 'fetch-network-policies',
  POD_SECURITY_POLICIES = 'fetch-pod-security-policies',
  CLUSTERS = 'fetch-clusters',
  BUILD_CLUSTER_RESOURCES_RELATIONSHIPS = 'build-cluster-resources-relationships',
  BUILD_CLUSTER_CLOUD_PROVIDERS_RELATIONSHIPS = 'build-cluster-cloud-providers-relationships',
  NAMESPACES = 'fetch-namespaces',
  NODES = 'fetch-nodes',
  SERVICES = 'fetch-services',
  DEPLOYMENTS = 'fetch-deployments',
  REPLICASETS = 'fetch-replica-sets',
  STATEFULSETS = 'fetch-stateful-sets',
  DAEMONSETS = 'fetch-daemon-sets',
  JOBS = 'fetch-jobs',
  CRONJOBS = 'fetch-cron-jobs',
  CONFIGMAPS = 'fetch-config-maps',
  SECRETS = 'fetch-secrets',
  PODS = 'fetch-pods',
}

export const Entities: Record<
  | 'NETWORK_POLICY'
  | 'POD_SECURITY_POLICY'
  | 'CLUSTER'
  | 'NAMESPACE'
  | 'NODE'
  | 'SERVICE'
  | 'DEPLOYMENT'
  | 'CONTAINER_SPEC'
  | 'REPLICASET'
  | 'STATEFULSET'
  | 'DAEMONSET'
  | 'JOB'
  | 'CRONJOB'
  | 'CONFIGMAP'
  | 'SECRET'
  | 'POD'
  | 'CONTAINER',
  StepEntityMetadata
> = {
  NETWORK_POLICY: {
    _type: 'kube_network_policy',
    _class: ['Configuration'],
    resourceName: 'Kubernetes Network Policy',
  },
  POD_SECURITY_POLICY: {
    _type: 'kube_pod_security_policy',
    _class: ['Configuration'],
    resourceName: 'Kubernetes Pod Security Policy',
  },
  CLUSTER: {
    _type: 'kube_cluster',
    _class: ['Cluster'],
    resourceName: 'Kubernetes Cluster',
  },
  NAMESPACE: {
    _type: 'kube_namespace',
    _class: ['Group'],
    resourceName: 'Kubernetes Namespace',
  },
  DEPLOYMENT: {
    _type: 'kube_deployment',
    _class: ['Deployment'],
    resourceName: 'Kubernetes Deployment',
  },
  CONTAINER_SPEC: {
    _type: 'kube_container_spec',
    _class: ['Configuration'],
    resourceName: 'Kubernetes Container Spec',
  },
  REPLICASET: {
    _type: 'kube_replica_set',
    _class: ['Deployment'],
    resourceName: 'Kubernetes ReplicaSet',
  },
  STATEFULSET: {
    _type: 'kube_stateful_set',
    _class: ['Deployment'],
    resourceName: 'Kubernetes StatefulSet',
  },
  DAEMONSET: {
    _type: 'kube_daemon_set',
    _class: ['Deployment'],
    resourceName: 'Kubernetes DaemonSet',
  },
  NODE: {
    _type: 'kube_node',
    _class: ['Host'],
    resourceName: 'Kubernetes Node',
  },
  JOB: {
    _type: 'kube_job',
    _class: ['Task'],
    resourceName: 'Kubernetes Job',
  },
  CRONJOB: {
    _type: 'kube_cron_job',
    _class: ['Task'],
    resourceName: 'Kubernetes CronJob',
  },
  CONFIGMAP: {
    _type: 'kube_config_map',
    _class: ['Configuration'],
    resourceName: 'Kubernetes ConfigMap',
  },
  SECRET: {
    _type: 'kube_secret',
    _class: ['Vault'],
    resourceName: 'Kubernetes Secret',
  },
  POD: {
    _type: 'kube_pod',
    _class: ['Task'],
    resourceName: 'Kubernetes Pod',
  },
  CONTAINER: {
    _type: 'kube_container',
    _class: ['Container'],
    resourceName: 'Kubernetes Container',
  },
  SERVICE: {
    _type: 'kube_service',
    _class: ['Service'],
    resourceName: 'Kubernetes Service',
  },
};

export const Relationships: Record<
  | 'CLUSTER_CONTAINS_POD_SECURITY_POLICY'
  | 'CLUSTER_CONTAINS_NAMESPACE'
  | 'CLUSTER_IS_AKS_CLUSTER'
  | 'NAMESPACE_CONTAINS_POD'
  | 'NAMESPACE_CONTAINS_SERVICE'
  | 'NAMESPACE_CONTAINS_DEPLOYMENT'
  | 'NAMESPACE_CONTAINS_REPLICASET'
  | 'NAMESPACE_CONTAINS_STATEFULSET'
  | 'NAMESPACE_CONTAINS_DAEMONSET'
  | 'NAMESPACE_CONTAINS_CRONJOB'
  | 'NAMESPACE_CONTAINS_JOB'
  | 'NAMESPACE_CONTAINS_CONFIGMAP'
  | 'NAMESPACE_CONTAINS_SECRET'
  | 'NAMESPACE_CONTAINS_NETWORK_POLICY'
  | 'DEPLOYMENT_MANAGES_REPLICASET'
  | 'DEPLOYMENT_USES_CONTAINER_SPEC'
  | 'REPLICASET_MANAGES_POD'
  | 'STATEFULSET_MANAGES_POD'
  | 'CRONJOB_MANAGES_JOB'
  | 'JOB_MANAGES_POD'
  | 'NODE_HAS_POD'
  | 'POD_CONTAINS_CONTAINER',
  StepRelationshipMetadata
> = {
  CLUSTER_CONTAINS_POD_SECURITY_POLICY: {
    _type: 'kube_cluster_contains_pod_security_policy',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.CLUSTER._type,
    targetType: Entities.POD_SECURITY_POLICY._type,
  },
  CLUSTER_CONTAINS_NAMESPACE: {
    _type: 'kube_cluster_contains_namespace',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.CLUSTER._type,
    targetType: Entities.NAMESPACE._type,
  },
  CLUSTER_IS_AKS_CLUSTER: {
    _type: 'kube_cluster_is_cluster',
    _class: RelationshipClass.IS,
    sourceType: Entities.CLUSTER._type,
    targetType: Entities.CLUSTER._type,
  },
  NAMESPACE_CONTAINS_POD: {
    _type: 'kube_namespace_contains_pod',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.POD._type,
  },
  NAMESPACE_CONTAINS_SERVICE: {
    _type: 'kube_namespace_contains_service',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.SERVICE._type,
  },
  NAMESPACE_CONTAINS_DEPLOYMENT: {
    _type: 'kube_namespace_contains_deployment',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.DEPLOYMENT._type,
  },
  NAMESPACE_CONTAINS_REPLICASET: {
    _type: 'kube_namespace_contains_replica_set',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.REPLICASET._type,
  },
  NAMESPACE_CONTAINS_STATEFULSET: {
    _type: 'kube_namespace_contains_stateful_set',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.STATEFULSET._type,
  },
  NAMESPACE_CONTAINS_DAEMONSET: {
    _type: 'kube_namespace_contains_daemon_set',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.DAEMONSET._type,
  },
  NAMESPACE_CONTAINS_CRONJOB: {
    _type: 'kube_namespace_contains_cron_job',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.CRONJOB._type,
  },
  NAMESPACE_CONTAINS_JOB: {
    _type: 'kube_namespace_contains_job',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.JOB._type,
  },
  NAMESPACE_CONTAINS_CONFIGMAP: {
    _type: 'kube_namespace_contains_config_map',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.CONFIGMAP._type,
  },
  NAMESPACE_CONTAINS_SECRET: {
    _type: 'kube_namespace_contains_secret',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.SECRET._type,
  },
  NAMESPACE_CONTAINS_NETWORK_POLICY: {
    _type: 'kube_namespace_contains_network_policy',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.NAMESPACE._type,
    targetType: Entities.NETWORK_POLICY._type,
  },
  DEPLOYMENT_MANAGES_REPLICASET: {
    _type: 'kube_deployment_manages_replica_set',
    _class: RelationshipClass.MANAGES,
    sourceType: Entities.DEPLOYMENT._type,
    targetType: Entities.REPLICASET._type,
  },
  DEPLOYMENT_USES_CONTAINER_SPEC: {
    _type: 'kube_deployment_uses_container_spec',
    _class: RelationshipClass.USES,
    sourceType: Entities.DEPLOYMENT._type,
    targetType: Entities.CONTAINER_SPEC._type,
  },
  REPLICASET_MANAGES_POD: {
    _type: 'kube_replica_set_manages_pod',
    _class: RelationshipClass.MANAGES,
    sourceType: Entities.REPLICASET._type,
    targetType: Entities.POD._type,
  },
  STATEFULSET_MANAGES_POD: {
    _type: 'kube_stateful_set_manages_pod',
    _class: RelationshipClass.MANAGES,
    sourceType: Entities.STATEFULSET._type,
    targetType: Entities.POD._type,
  },
  CRONJOB_MANAGES_JOB: {
    _type: 'kube_cron_job_manages_job',
    _class: RelationshipClass.MANAGES,
    sourceType: Entities.CRONJOB._type,
    targetType: Entities.JOB._type,
  },
  JOB_MANAGES_POD: {
    _type: 'kube_job_manages_pod',
    _class: RelationshipClass.MANAGES,
    sourceType: Entities.JOB._type,
    targetType: Entities.POD._type,
  },
  NODE_HAS_POD: {
    _type: 'kube_node_has_pod',
    _class: RelationshipClass.HAS,
    sourceType: Entities.NODE._type,
    targetType: Entities.POD._type,
  },
  POD_CONTAINS_CONTAINER: {
    _type: 'kube_pod_contains_container',
    _class: RelationshipClass.CONTAINS,
    sourceType: Entities.POD._type,
    targetType: Entities.CONTAINER._type,
  },
};
