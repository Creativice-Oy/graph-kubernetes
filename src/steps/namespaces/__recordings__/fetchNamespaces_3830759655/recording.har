{
  "log": {
    "_recordingName": "fetchNamespaces",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "9d73307742273594a28ec2e4efce503e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 120,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "50"
            }
          ],
          "url": "https://192.168.49.2:8443/api/v1/namespaces?limit=50"
        },
        "response": {
          "bodySize": 1658,
          "content": {
            "mimeType": "application/json",
            "size": 1658,
            "text": "{\"kind\":\"NamespaceList\",\"apiVersion\":\"v1\",\"metadata\":{\"resourceVersion\":\"575619\"},\"items\":[{\"metadata\":{\"name\":\"default\",\"uid\":\"d5976ece-dc03-46a1-921f-cb6043e5f56e\",\"resourceVersion\":\"200\",\"creationTimestamp\":\"2021-04-27T20:57:19Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-04-27T20:57:19Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-node-lease\",\"uid\":\"55ea52dd-382c-4b8c-8c24-bff59db4b817\",\"resourceVersion\":\"66\",\"creationTimestamp\":\"2021-04-27T20:57:17Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-04-27T20:57:17Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-public\",\"uid\":\"35ca0e47-73b0-4e3d-9e2a-d8f3d6506d88\",\"resourceVersion\":\"58\",\"creationTimestamp\":\"2021-04-27T20:57:17Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-04-27T20:57:17Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-system\",\"uid\":\"239ee6a3-23d6-4ca6-ae32-3b3fdda6a604\",\"resourceVersion\":\"34\",\"creationTimestamp\":\"2021-04-27T20:57:17Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-04-27T20:57:17Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "210417ef-0504-45ee-ba4b-365d19207515"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "f6528105-041a-490a-9dfb-a86e0c629d2a"
            },
            {
              "name": "date",
              "value": "Thu, 24 Jun 2021 18:35:16 GMT"
            },
            {
              "name": "content-length",
              "value": "1658"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 289,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-24T18:35:16.394Z",
        "time": 28,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 28
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
