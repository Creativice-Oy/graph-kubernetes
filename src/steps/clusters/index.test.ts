import { createDataCollectionTest } from '../../../test/recording';
import { buildClusterResourcesRelationships, fetchClusterDetails } from '.';
import { integrationConfig } from '../../../test/config';
import { Entities, Relationships } from '../constants';
import { fetchNamespaces } from '../namespaces';
import { RelationshipClass } from '@jupiterone/data-model';

describe('#fetchClusterDetails', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'fetchClusterDetails',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchClusterDetails],
      entitySchemaMatchers: [
        {
          _type: Entities.CLUSTER._type,
          matcher: {
            _class: ['Cluster'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'kube_cluster' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                name: { type: 'string' },
                displayName: { type: 'string' },
              },
            },
          },
        },
      ],
    });
  });
});

describe('#buildClusterResourcesRelationships', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'buildClusterResourcesRelationships',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [
        fetchClusterDetails,
        fetchNamespaces,
        buildClusterResourcesRelationships,
      ],
      entitySchemaMatchers: [],
      relationshipSchemaMatchers: [
        {
          _type: Relationships.CLUSTER_CONTAINS_NAMESPACE._type,
          matcher: {
            schema: {
              properties: {
                _class: { const: RelationshipClass.CONTAINS },
                _type: { const: 'kube_cluster_contains_namespace' },
              },
            },
          },
        },
      ],
    });
  });
});
