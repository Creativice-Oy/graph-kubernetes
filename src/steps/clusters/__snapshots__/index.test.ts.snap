// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#buildClusterResourcesRelationships should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Cluster",
      ],
      "_key": "kube_cluster:local-integration-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "name": "Local Integration",
          },
        },
      ],
      "_type": "kube_cluster",
      "createdOn": undefined,
      "displayName": "Local Integration",
      "name": "Local Integration",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "9c533ce3-b6bd-43e6-947a-f9f73dc297b5",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-05-13T19:34:38.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-05-13T19:34:38.000Z,
                },
              ],
              "name": "default",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "155",
              "selfLink": "/api/v1/namespaces/default",
              "uid": "9c533ce3-b6bd-43e6-947a-f9f73dc297b5",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1620934478000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "default",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "default",
      "resourceVersion": "155",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "11a8778d-4ef0-437b-bb1d-db3cf0a26e82",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-05-13T19:34:36.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-05-13T19:34:36.000Z,
                },
              ],
              "name": "kube-node-lease",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "33",
              "selfLink": "/api/v1/namespaces/kube-node-lease",
              "uid": "11a8778d-4ef0-437b-bb1d-db3cf0a26e82",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1620934476000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-node-lease",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-node-lease",
      "resourceVersion": "33",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "ba4e6047-864a-46f9-9f29-3ffe004d7f13",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-05-13T19:34:36.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-05-13T19:34:36.000Z,
                },
              ],
              "name": "kube-public",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "25",
              "selfLink": "/api/v1/namespaces/kube-public",
              "uid": "ba4e6047-864a-46f9-9f29-3ffe004d7f13",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1620934476000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-public",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-public",
      "resourceVersion": "25",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "de2a4eee-505b-49d7-bc08-09026c88dae5",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-05-13T19:34:36.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-05-13T19:34:36.000Z,
                },
              ],
              "name": "kube-system",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "16",
              "selfLink": "/api/v1/namespaces/kube-system",
              "uid": "de2a4eee-505b-49d7-bc08-09026c88dae5",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1620934476000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-system",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-system",
      "resourceVersion": "16",
      "status.phase": "Active",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "kube_cluster:local-integration-instance",
      "_key": "kube_cluster:local-integration-instance|contains|9c533ce3-b6bd-43e6-947a-f9f73dc297b5",
      "_toEntityKey": "9c533ce3-b6bd-43e6-947a-f9f73dc297b5",
      "_type": "kube_cluster_contains_namespace",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "kube_cluster:local-integration-instance",
      "_key": "kube_cluster:local-integration-instance|contains|11a8778d-4ef0-437b-bb1d-db3cf0a26e82",
      "_toEntityKey": "11a8778d-4ef0-437b-bb1d-db3cf0a26e82",
      "_type": "kube_cluster_contains_namespace",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "kube_cluster:local-integration-instance",
      "_key": "kube_cluster:local-integration-instance|contains|ba4e6047-864a-46f9-9f29-3ffe004d7f13",
      "_toEntityKey": "ba4e6047-864a-46f9-9f29-3ffe004d7f13",
      "_type": "kube_cluster_contains_namespace",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "kube_cluster:local-integration-instance",
      "_key": "kube_cluster:local-integration-instance|contains|de2a4eee-505b-49d7-bc08-09026c88dae5",
      "_toEntityKey": "de2a4eee-505b-49d7-bc08-09026c88dae5",
      "_type": "kube_cluster_contains_namespace",
      "displayName": "CONTAINS",
    },
  ],
  "encounteredTypes": Array [
    "kube_cluster",
    "kube_namespace",
    "kube_cluster_contains_namespace",
  ],
  "numCollectedEntities": 5,
  "numCollectedRelationships": 4,
}
`;

exports[`#fetchClusterDetails should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Cluster",
      ],
      "_key": "kube_cluster:local-integration-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "name": "Local Integration",
          },
        },
      ],
      "_type": "kube_cluster",
      "createdOn": undefined,
      "displayName": "Local Integration",
      "name": "Local Integration",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "kube_cluster",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
